integration-tests/features/steps/common.py

```diff
@@ -190,6 +191,11 @@ def finish_analysis_for_component(context, ecosystem, component, version):
         raise Exception('Timeout waiting for the component analysis results')
 
 
+def contains_alternate_node(json_resp):
+    """Check for the existence of alternate node in the stack analysis."""
+    return json_resp['result'][0].get('recommendations', None).get('alternate', None) is not None
```

Comment:
If there's no `'recommendations'`, the second `get()` fails, because the first one returns `None`.
suggestion: `.get('recommendations', {})`

If I wanted to be picky, I'd also say that `json_resp['result'][0]` fails if `json_resp['result']` is empty or there's even no `'result'` in `json_resp`.
suggestion: `return bool(json_resp.get('result')) and json_resp['result'][0].get('recommendations', {}).get('alternate', None) is not None`